modules/basics/src/main/java/com/opengamma/basics/currency/FxMatrix.java

```diff
@@ -0,0 +1,557 @@
+/**
+ * Copyright (C) 2011 - present by OpenGamma Inc. and the OpenGamma group of companies
+ *
+ * Please see distribution for license.
+ */
+package com.opengamma.basics.currency;
+
+import static com.opengamma.collect.Guavate.entriesToImmutableMap;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.BiConsumer;
+import java.util.stream.Collector;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import com.opengamma.collect.ArgChecker;
+import com.opengamma.collect.tuple.Pair;
+
+/**
+ * Immutable class describing a set of currencies and all the cross rates between them.
+ */
+public class FxMatrix {
+
+  /**
+   * An FX matrix containing neither currencies nor rates.
+   */
+  public static final FxMatrix EMPTY_FX_MATRIX = builder().build();
+
+  /**
+   * The map between the currencies and their order. An ImmutableMap is
```

Comment:
Elaborate on what this is for. When I read it I thought "what order?". It was only when I read the Javadoc below that I realised it's the index of the currencies row in ``rates``.